### PR TITLE
HARP-15180 Upgrading yeoman because it has a transitive dependency (axios) which is out of date

### PR DESCRIPTION
**Pull request by nzjony**
Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/generator-harp.gl/test/test.js [hunk 1]
```diff
@@ -17,34 +13,35 @@ const {
...
+            .run(path.join(__dirname, "../generators/app"))
             .inTmpDir(function (dir) {
                 if (process.env.USE_NPMRC) {
                     const targetPath = path.join(dir, ".npmrc");
                     fs.copyFileSync(process.env.USE_NPMRC, targetPath);
                 }
             })
             .withPrompts({
-                'access_token': 'test_token'
+                language: "typescript",
```
Comment (nzjony) [inline @here/generator-harp.gl/test/test.js:30]: these match what is queried in the Generator, i.e. language & apikey. The `packagename` already has a default, so this isn't required.

File: @here/generator-harp.gl/test/test.js [hunk 2]
```diff
@@ -17,34 +13,35 @@ const {
...
+                apikey: "test_token"
             });
 
-        assert.isString(helperDirectory);
+        assert.isString(helperDirectory.cwd);
```
Comment (dbacarel) [inline @here/generator-harp.gl/test/test.js:34]: `cwd` is not public according to https://yeoman.github.io/yeoman-test/YeomanTest.html, we might need to use a separate variable.
Can we assume that the current cwd is `path.join(__dirname, "../generators/app")` ?

**Review by dbacarel — CHANGES_REQUESTED** [review]

File: @here/generator-harp.gl/package.json [hunk 3]
```diff
@@ -23,13 +23,14 @@
...
         "mkdirp": "^1.0.4",
-        "yeoman-generator": "^4.10.1"
+        "yeoman-generator": "^5.2.0"
     },
     "devDependencies": {
         "chai": "^4.2.0",
         "fs.extra": "^1.3.2",
         "mocha": "^8.2.1",
-        "yeoman-test": "^2.6.0"
+        "yeoman-environment": "^3.4.0",
```
[the remark this answers is not in the document]
Comment (nzjony) [reply @here/generator-harp.gl/package.json:32]: Good spotting, thank you.

File: @here/generator-harp.gl/test/test.js (hunk 2 above)
Comment (nzjony) [reply @here/generator-harp.gl/test/test.js:34]: Good point, yes I guess that makes sense.

Actually checking the `isString` is anyway some what ambiguous, I will check that the run result is not undefined.

Comment (nzjony) [reply @here/generator-harp.gl/test/test.js]: I see the code: https://yeoman.github.io/yeoman-test/run-result.js.html

And there is a `cwd`, so whilst not part of the documentation, it is part of the public API, at least according to the code.

We call `inTmpDir`, which gives us some random tmp directory, so it is different that the path.join you mentioned and not so easy to get the `cwd`, at least I don't see a method to return it.

**Review by dbacarel — APPROVED** [review]

**Pull request merged**